third_party/terraform/data_sources/data_source_secretmanager_secret_version.go.erb

```diff
@@ -0,0 +1,122 @@
+<% autogen_exception -%>
+package google
+<% unless version == "ga" -%>
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceSecretmanagerSecretVersion() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSecretmanagerSecretVersionRead,
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"secret": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"destroy_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"secret_data": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func dataSourceSecretmanagerSecretVersionRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	fv, err := parseProjectFieldValue("secrets", d.Get("secret").(string), "project", d, config, false)
+	if err != nil {
+		return err
+	}
+	project := fv.Project
+	d.Set("secret", fv.Name)
+
+	var url string
+	versionNum := d.Get("version")
+
+	if versionNum != "" {
+		url, err = replaceVars(d, config, "{{SecretmanagerBasePath}}projects/{{project}}/secrets/{{secret}}/versions/{{version}}")
```

Comment:
These will always pick up the default project config rather than the project contained in `secret` if it is a long name?

If the config looks like:
```
provider {
  project = "my-other-project"
}

"google_secret_version" "my-secret" {
  secret = "projects/my-project1/secrets/my-secret"
}
```

We will pick up project as "my-other-project" I believe

Reply:
Ah, yes, that makes sense.  Should I `d.Set("project", fv.Project)` up by line 65 then?

Reply:
But then we will override an explicit `project` set on the `secret_version` itself. Like:
```
"google_secret_version" "my-secret" {
  project = "my-other-project"
  secret = "projects/my-project1/secrets/my-secret"
}
```

But that might be ok? Maybe check `d.Get("project")` and if set & not equal to `fv.Project` then throw an error. Otherwise `d.Set("project", fv.Project)`?